chore(template): improve PullRequest template

ClxUne09 opened the pull request:
### Prerequisites

- [x] I have read and understood the [contributing guide][CONTRIBUTING.md].
- [x] The commit message follows the [conventional commits][cc] guidelines.
- [x] Tests for the changes have been added (for bug fixes / features).
- [x] Docs have been added/updated (for bug fixes / features).

### Description


- Added sections for clear description including summary, problem, solution, and impact.
- Updated Helpful Resources section with links to contributing guides, Git practices, and community standards.

---

### 05e491b641baeda3c0d5d41d3ed99d9efac85328:

1. **Commitlint Rules:** Added a reference to commitlint rules and standards to ensure that commit messages follow guidelines.
2. **Extended Description:** Expanded the description section to include an additional point for indicating which issues the PR resolves or closes.

These updates aim to enhance clarity and ensure that pull requests adhere to our contribution and commit message standards.

[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
[cc]: https://www.conventionalcommits.org/en/v1.0.0/#summary

ClxUne09 committed:
chore(template): improve PullRequest template


- Added sections for clear description including summary, problem, solution, and impact.
- Updated Helpful Resources section with links to contributing guides, Git practices, and community standards.

lewis-yeung commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,59 @@
+- [ ] I have read and understood the [Contributing Guide][contributing].
+- [ ] My commit message follows the [Conventional Commits][cc] guidelines.
+- [ ] Relevant tests have been added or updated (for bug fixes/features).
+- [ ] Documentation has been added or updated (for bug fixes/features).
```
```suggestion
- [ ] Relevant tests have been added or updated (for bug fixes or features).
- [ ] Documentation has been added or updated (for bug fixes or features).
```

Instead of separating phrases with a slash, I think using an "or" is clearer. Just a small suggestion. :)

ClxUne09 committed:
style(template): refine PullRequest template for clarity

Co-authored-by: L. Yeung <lewis-yeung@foxmail.com>

ClxUne09 committed:
Merge branch 'main' into pr-templ

ClxUne09 committed:
chore(template): update PullRequest template


1. **Commitlint Rules:** Added a reference to commitlint rules and standards to ensure that commit messages follow guidelines.
2. **Extended Description:** Expanded the description section to include an additional point for indicating which issues the PR resolves or closes.

These updates aim to enhance clarity and ensure that pull requests adhere to our contribution and commit message standards.

ClxUne09 committed:
Merge branch 'main' into pr-templ

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
-- [ ] Tests for the changes have been added (for bug fixes / features).
-- [ ] Docs have been added/updated (for bug fixes / features).
+- [ ] I have read and understood the [Contributing Guide][contributing].
+- [ ] My commit messages follow the [Conventional Commits][cc] guidelines and our [commitlint rules and standards][commitlint].
```
commitlint enforces conventional commits, so I find that phrase rather strange.

```suggestion
- [ ] My commit messages are written according the [Conventional Commits][cc] guidelines, respecting our [rules][commitlint].
```

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
-If you're not comfortable with working with Git,
-we're working a guide (https://ohmyposh.dev/docs/contributing/git) to help you out.
-Oh My Posh advises GitKraken (https://www.gitkraken.com/invite/nQmDPR9D) as your preferred cross platform Git GUI power tool.
+Please provide a clear and concise description of the changes introduced by this pull request. You should explain the following:
```
```suggestion
Please provide a clear and concise description of the changes introduced by this pull request. Explain the following when relevant:
```

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
-[CONTRIBUTING.md]: https://github.com/JanDeDobbeleer/oh-my-posh/blob/main/CONTRIBUTING.md
+---
+
+<br>
```
why are the linebreaks here?

ClxUne09 replied:
The line breaks are used to emphasize the main PR description section, making it easier to focus on providing a clear and concise overview of the changes.

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
+- **[Oh My Posh Git Guide][git-guide]:** For understanding Git practices and commands.
+- **[Discussions Page][discussions]:** Join project discussions and ask questions.
+- **[GitKraken Guide][gitkraken]:** Recommended tool for managing Git repositories.
+<!-- - **[Code Structure Guide][code-structure]:** Understand the overall organization of our codebase. -->
```
this isn't relevant here

ClxUne09 replied:
I think it's somewhat related to the pull request. It helps contributors better understand the PR.
I think adding this section to the docs is a good idea; we can link to the page we’ll create in the template. We can use this interim solution until the docs section is finalized.

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
+
+---
+
+### Helpful Resources
```
there's a reason this was hidden, we shouldn't see this after creation

ClxUne09 replied:
Are you sure you want to comment that? It has a styling aspect for the preview, but it will also take up some space.

JanDeDobbeleer commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,21 +1,72 @@
+- **[Community Standard][community]:** Engage with the community and understand our community standards.
+- **[Contributing Guide][contributing]:** Read and understand our guidelines for contributing.
+- **[GitHub’s Pull Request Review Guide][gh-pr-review]:** Additional guidance on the review process.
+<!-- - **[Pitfalls to Avoid][pitfalls]:** Common mistakes to watch out for when contributing. -->
```
this isn't relevant here

ClxUne09 replied:
I think it's somewhat related to the pull request. It helps contributors better understand the PR.
I think adding this section to the docs is a good idea; we can link to the page we’ll create in the template. We can use this interim solution until the docs section is finalized.

ClxUne09 committed:
refactor(template): update PR commit message guidelines

Co-authored-by: Jan De Dobbeleer <2492783+JanDeDobbeleer@users.noreply.github.com>

ClxUne09 committed:
refactor(template): clarify requested description for PR

Co-authored-by: Jan De Dobbeleer <2492783+JanDeDobbeleer@users.noreply.github.com>

ClxUne09 commented:
Did I make a mistake that I should fix?

ClxUne09 committed:
Merge branch 'main' into pr-templ

JanDeDobbeleer commented:
> Did I make a mistake that I should fix?

@ClxUne09 I was in doubt if I should merge this. I wasn't immediately convinced by the guidelines as I'm not really missing anything usually. When I get that feeling I let things marinate a bit, but TBH, I still feel the same.

ClxUne09 commented:
> > Did I make a mistake that I should fix?
> 
> \@ClxUne09 I was in doubt if I should merge this. I wasn't immediately convinced by the guidelines as I'm not really missing anything usually. When I get that feeling I let things marinate a bit, but TBH, I still feel the same.

You’re right; these changes focus more on the details rather than the actual parts. I will close this pull request for now. Thank you for your reply.
